packages/components/form/src/form.vue

```diff
@@ -1,258 +1,158 @@
-    const index = getLabelWidthIndex(val)
-    index > -1 && potentialLabelWidthArr.value.splice(index, 1)
+const removeField: FormContext['removeField'] = (field) => {
+  if (!field.prop) {
+    fields.splice(fields.indexOf(field), 1)
   }
+}
 
-  return {
-    autoLabelWidth,
-    registerLabelWidth,
-    deregisterLabelWidth,
+const resetFields: FormContext['resetFields'] = (properties = []) => {
+  if (!props.model) {
+    debugWarn(COMPONENT_NAME, 'model is required for resetFields to work.')
+    return
   }
+  filterFields(fields, properties).forEach((field) => field.resetField())
 }
 
-export interface Callback {
-  (isValid?: boolean, invalidFields?: ValidateFieldsError): void
+const clearValidate: FormContext['clearValidate'] = (props = []) => {
+  filterFields(fields, props).forEach((field) => field.clearValidate())
 }
 
-export default defineComponent({
-  name: 'ElForm',
-  props: {
-    model: Object,
-    rules: Object as PropType<FormRulesMap>,
-    labelPosition: String,
-    labelWidth: {
-      type: [String, Number],
-      default: '',
-    },
-    labelSuffix: {
-      type: String,
-      default: '',
-    },
-    inline: Boolean,
-    inlineMessage: Boolean,
-    statusIcon: Boolean,
-    showMessage: {
-      type: Boolean,
-      default: true,
-    },
-    size: String as PropType<ComponentSize>,
-    disabled: Boolean,
-    validateOnRuleChange: {
-      type: Boolean,
-      default: true,
-    },
-    hideRequiredAsterisk: {
-      type: Boolean,
-      default: false,
-    },
-    scrollToError: Boolean,
-  },
-  emits: ['validate'],
-  setup(props, { emit }) {
-    const fields: FormItemCtx[] = []
-
-    watch(
-      () => props.rules,
-      () => {
-        fields.forEach((field) => {
-          field.evaluateValidationEnabled()
-        })
+const validate = async (callback?: FormValidateCallback): Promise<void> =>
+  validateField(undefined, callback)
+
+const validateField: FormContext['validateField'] = async (
+  properties = [],
+  callback
+) => {
+  if (callback) {
+    validate()
+      .then(() => callback(true))
+      .catch((fields: ValidateFieldsError) => callback(false, fields))
+    return
+  }
 
-        if (props.validateOnRuleChange) {
-          validate(() => ({}))
-        }
-      }
-    )
+  const { model, scrollToError } = props
 
-    const formSize = useSize()
-    const prefix = 'el-form'
-    const formKls = computed(() => {
-      const { labelPosition, inline } = props
-      return [
-        prefix,
-        `${prefix}--${formSize.value}`,
-        labelPosition ? `${prefix}--label-${labelPosition}` : '',
-        inline ? `${prefix}--inline` : '',
-      ]
-    })
+  if (!model) {
+    debugWarn(COMPONENT_NAME, 'model is required for validate to work!')
+    return
+  }
+  if (fields.length === 0) {
+    return
+  }
 
-    const addField = (field: FormItemCtx) => {
-      if (field) {
-        fields.push(field)
-      }
-    }
+  const fds = filterFields(fields, properties)
```

Comment:
**Suggestion:** name this to `filteredFields`, do not shorten the variable name because the bundler will do that for you. These variables are meant to be seen by human so it should be human readable.